recirq/qml_lfe/sample_analysis.ipynb

```diff
@@ -0,0 +1,1451 @@
+    "                \n",
+    "                val *= 1 / (p1 + q1 - 1)\n",
+    "                \n",
+    "                a[i] = int(a[i])\n",
+    "                b1 = int(a[i] // 2)\n",
+    "                s1 = int(a[i] % 2)\n",
+    "\n",
+    "                if s1 == 0:\n",
+    "                    if random.choices([0, 1], weights = [q1, 1-q1], k=1)[0] == 1:\n",
+    "                        s1 = 1 - s1\n",
+    "                        val *= -1\n",
+    "                else:\n",
+    "                    if random.choices([0, 1], weights = [p1, 1-p1], k=1)[0] == 1:\n",
+    "                        s1 = 1 - s1\n",
+    "                        val *= -1\n",
+    "\n",
+    "                if idx_to_pauli[b1] == pauli:\n",
+    "                    val *= (1 if s1 == 0 else -1)\n",
+    "                else:\n",
+    "                    val = 0\n",
+    "            \n",
+    "            if val != 0:\n",
+    "                ttl_sum += val\n",
+    "                ttl_cnt += 1\n",
+    "    return (ttl_sum / ttl_cnt if ttl_cnt > 0 else None)\n",
+    "\n",
+    "mypath = \"data/\"\n",
+    "\n",
+    "n = 8\n",
+    "number_of_experiment = 500\n",
+    "correct_cnt, total_cnt = 0, 0\n",
+    "\n",
+    "# Noiseless simulation (measurement error = 0)\n",
+    "calib_2x2 = []\n",
+    "for i in range(2*n):\n",
+    "    conf_mtx = np.zeros((2, 2))\n",
+    "    conf_mtx[0][0] = 1\n",
+    "    conf_mtx[0][1] = 0\n",
+    "    conf_mtx[1][0] = 0\n",
+    "    conf_mtx[1][1] = 1\n",
+    "    calib_2x2.append(conf_mtx)\n",
+    "\n",
+    "onlyfiles = [f for f in listdir(mypath) if isfile(join(mypath, f))]\n",
+    "\n",
+    "for file in onlyfiles * 5:\n",
+    "    splt = file.split(\"-\")\n",
+    "\n",
+    "    if splt[0] != \"C\": continue\n",
+    "    if int(splt[2]) != n: continue\n",
+    "    if len(splt) != 5: continue\n",
+    "\n",
+    "    pauli = splt[-1][:-4]\n",
+    "\n",
+    "    binaryout = np.load(join(mypath, file))\n",
+    "    basisout = np.load(join(mypath, \"-\".join(splt[:-1]) + \"-\" + splt[-1][:-4] + \"-basis.npy\"))\n",
+    "    fulldata = basisout*2 + binaryout\n",
+    "\n",
+    "    data = np.random.permutation(fulldata)[:number_of_experiment]\n",
+    "\n",
+    "    rand_P = [random.choice([\"I\", \"X\", \"Y\", \"Z\"]) for i in range(n)]\n",
+    "    while min([x == y for (x, y) in zip(pauli, rand_P)]) == 1 or sum([p == \"I\" for p in rand_P]) == (n):\n",
+    "        rand_P = [random.choice([\"I\", \"X\", \"Y\", \"Z\"]) for i in range(n)]\n",
+    "\n",
+    "    pred1 = predict_exp(data, pauli, calib_2x2)\n",
+    "    pred2 = predict_exp(data, rand_P, calib_2x2)\n",
+    "    \n",
+    "    if pred1 is None or pred2 is None:\n",
+    "        correct_cnt += 0.5\n",
+    "    elif abs(pred1) > abs(pred2):\n",
+    "        correct_cnt += 1\n",
+    "    total_cnt += 1\n",
+    "\n",
+    "print(\"ACC:\", correct_cnt / total_cnt)\n"
```

Comment:
I really don't understand what's going on here.  Is ACC:0.465 good, bad, what does it mean?  This whole code block really needs more explanation to be useful.